rust/src/verify.rs

```diff
@@ -270,13 +270,18 @@ fn left_branches_are_empty(
     branch: i32,
 ) -> Result<bool> {
     if let Some(&idx) = spec.child_order.iter().find(|&&x| x == branch) {
-        // compare the prefix bytes with the appropriate number of empty children
-        let left_children = spec.child_order.len() - 1 - idx as usize;
+        // count branches to left of this
+        let left_branches = idx as usize;
+        if left_branches == 0 {
+            return Ok(false);
```

Comment:
I disagree with this for the same reason I disagree with the changes to the Go logic above.

left_branches == 0 not only satisfies what I think the function should mean but also the function comment in the Go code.